src/systems/lobby-camera-system.js

```diff
@@ -1,79 +1,8 @@
-const NETWORKED_LOBBY_CAMERA_STATE = "networked-lobby-camera-state";
-
-const setMatrixWorld = (function() {
-  const inv = new THREE.Matrix4();
-  return function setMatrixWorld(object3D, m) {
-    object3D.matrixWorld.copy(m);
-    object3D.matrix = inv.getInverse(object3D.parent.matrixWorld).multiply(object3D.matrixWorld);
-    object3D.matrix.decompose(object3D.position, object3D.quaternion, object3D.scale);
-  };
-})();
-
-AFRAME.registerComponent("lobby-camera-transform-component", {
-  init() {
-    this.el.id = NETWORKED_LOBBY_CAMERA_STATE;
-  },
-  tick() {
-    if (NAF.utils.isMine(this.el)) {
-      this.playerCamera = this.playerCamera || document.querySelector("#player-camera").object3D;
-      setMatrixWorld(this.el.object3D, this.playerCamera.matrixWorld);
-    }
-  }
-});
-
-export function takeOrCreateLobbyCamera() {
-  const state = document.getElementById(NETWORKED_LOBBY_CAMERA_STATE);
-  if (!state) {
-    const e = document.createElement("a-entity");
-    e.setAttribute("networked", "template: #lobby-camera");
-    e.id = NETWORKED_LOBBY_CAMERA_STATE;
-    AFRAME.scenes[0].appendChild(e);
-  } else {
-    NAF.utils.takeOwnership(state);
-  }
-}
-
-export function lobbyCameraIsMine() {
-  const state = document.getElementById(NETWORKED_LOBBY_CAMERA_STATE);
-  return state && NAF.utils.isMine(state);
-}
-
-export function releaseLobbyCamera() {
-  const state = document.getElementById(NETWORKED_LOBBY_CAMERA_STATE);
-  state.parentNode.removeChild(state);
-}
-
-function avatarForSessionId(sessionId) {
-  const avatars = document.querySelectorAll("[networked-avatar]");
-  for (let i = 0; i < avatars.length; i++) {
-    const avatar = avatars[i];
-    if (avatar.components.networked.data.owner === sessionId) {
-      return avatar;
-    }
-  }
-  return null; // Have not spawned an avatar yet, or invalid session id.
-}
-
 export class LobbyCameraSystem {
   tick() {
-    const state = document.getElementById(NETWORKED_LOBBY_CAMERA_STATE);
-    if (AFRAME.scenes[0].is("entered") || !state || !state.parentNode || NAF.utils.isMine(state)) {
-      return;
+    const el = document.querySelector("[scene-preview-camera]");
+    if (el) {
+      el.components["scene-preview-camera"].tick2();
```

Comment:
not a strong opinion here, but it does seem a bit weird to refactor this into this ECS pattern when there is just a single instance

Reply:
I didn't put this code here because I care about the ECS pattern. I only care that the tick happens after the `networked` component `tick`, which lerps the remote avatar's head transform. The same can be said of some of the other systems in `hubs-systems`. Some of them were written with an ECS pattern in mind but a lot of it is just about wanting a defined execution order.

Reply:
Open to suggestions for how else to accomplish the desired tick order. One way is to just change `scene-preview-camera` into an aframe system instead of this class with a `tick` method.

Reply:
ah that makes sense, thanks for clarifying -- seems OK to leave as-is.